src/handlers/edit.ts

```diff
@@ -0,0 +1,42 @@
+import { Task } from "../models/Task";
+import { getTasks, saveTasks } from "../services/tasks";
+
+export namespace EditHandler {
+  export type Args = {
+    id: string;
+    done: boolean;
+    pending: boolean;
+    description?: string;
+    by?: string;
+  };
+
+  export const handle = async (args: Args): Promise<void> => {
+    let allTasks: Array<Task> = await getTasks();
+
+    let task: Task | undefined = allTasks.find((task) => task.id === args.id);
+
+    if (task === undefined) {
+      console.error("This ID does not exist. Please type valid ID!");
+      process.exit(1);
+    }
+
+    if (typeof args.description === "string") {
+      task.description = args.description;
+    }
+
+    if (typeof args.by === "string") {
+      task.by = new Date(args.by).toISOString();
```

Comment:
What would happen if the user doesn't provide a valid date? E.g. `--by "tomorrow"`

Reply:
That's a good point and I didn't think about that! I made an issue for myself so that I can fix this later for other handlers as well (#13 )